src/containers/CompanyContainer.tsx

```diff
@@ -1,11 +1,169 @@
-import { Flex } from '@chakra-ui/react';
+import React from 'react';
 
-import { CompanyCreationForm } from '@components/company';
+import { Box, Flex, useToast, VStack, Text } from '@chakra-ui/react';
+
+import { CompanyCreationForm, DomainStatus } from '@components/company';
+
+import GoogleIconImage from 'assets/google.png';
+import NetlifyIconImage from 'assets/netlify.png';
+
+import { useAddDNSRecord } from 'hooks/apiHooks/company/useAddDNSRecord';
+import { useAddDomainAlias } from 'hooks/apiHooks/company/useAddDomainAlias';
+import { useCreateCompany } from 'hooks/apiHooks/company/useCreateCompany';
+
+import type { CompanyFormProps } from 'interfaces';
+
+const RETRY_LIMIT = 3;
 
 export const CompanyContainer = () => {
+    const toast = useToast();
+    const createCompany = useCreateCompany();
+    const addDNSRecord = useAddDNSRecord();
+    const addDomainAlias = useAddDomainAlias();
+
+    const [formCompanyId, setFormCompanyId] = React.useState('');
+    const [showDNSStatus, setShowDNSStatus] = React.useState(false);
+    const [showDomainAliasStatus, setShowDomainAliasStatus] =
+        React.useState(false);
+    const [dnsRetryCount, setDnsRetryCount] = React.useState(0);
+    const [domainAliasRetryCount, setDomainAliasRetryCount] = React.useState(0);
+
+    const isCreationStatusVisible = React.useMemo(() => {
+        return (
+            createCompany.isSuccess && showDNSStatus && showDomainAliasStatus
+        );
+    }, [createCompany.isSuccess, showDNSStatus, showDomainAliasStatus]);
+
+    const isCreatingCompany = React.useMemo(() => {
+        return (
+            createCompany.isLoading ||
+            addDNSRecord.isLoading ||
+            addDomainAlias.isLoading
+        );
+    }, [
+        addDNSRecord.isLoading,
+        addDomainAlias.isLoading,
+        createCompany.isLoading
+    ]);
+
+    const createDNSRecord = (companyId: string) => {
+        // TODO: Remove two lines below once testing is done
+        setShowDNSStatus(true);
+        return;
+
+        addDNSRecord.mutate(
+            {
+                params: { companyId },
+                requestBody: { name: `${companyId}.hunar.ai` }
+            },
+            {
+                onSettled: () => {
+                    setShowDNSStatus(true);
+                    setDnsRetryCount(count => count + 1);
+                }
+            }
+        );
+    };
+
+    const createDomainAlias = (companyId: string) => {
+        // TODO: Remove two lines below once testing is done
+        setShowDomainAliasStatus(true);
+        return;
```

Comment:
NOTE: these two lines are added for QA testing and will removed once QA is complete